droidlet/memory/sql_memory.py

```diff
@@ -1153,3 +1169,9 @@ def safe_unpickle(self, bs):
         obj = pickle.loads(bs)
         self.reinstate_attrs(obj)
         return obj
+
+    def register_hook(self, receiver, sender):
+        """
+        allows for registering hooks using the event dispatcher
+        """
+        self.dispatch_signal.connect(receiver, sender)
```

Comment:
you should verify that the sender can only be one of `self.db_write`, `self.db_read`, etc.

You shouldn't be registering hooks for any sender via `memory.register_hook`

Reply:
Will implement this right now, thanks for the feedback!